src/Kunstmaan/NodeBundle/Controller/NodeAdminController.php

```diff
@@ -1219,6 +1333,34 @@ private function deleteNodeChildren(
         }
     }
 
+    /**
+     * @param Node $node
+     */
+    private function undoDeleteNode(
+        Node $node
+    ) {
+        $this->denyAccessUnlessGranted(
```

Comment:
I would add a early return here if the node is not deleted. A possible case could be that you delete a parent with nested children and already restore a child before.

Reply:
I've applied this